tap_pardot/client.py

```diff
@@ -122,6 +125,17 @@ def _refresh_access_token(self):
         headers = {"Content-Type": "application/x-www-form-urlencoded"}
         response = self.requests_session.post(url, data=data, headers=headers)
         self.access_token = response.json().get("access_token")
+
+        response = self.requests_session.post(url, data=data, headers=headers)
+        data = response.json()
+        self.access_token = data.get("access_token")
+
+        if response.status_code == 400 and data.get("error") == "invalid_grant":
+            error = data.get("error")
+            message = data.get("error_description")
+            raise InvalidCredentials(f"Invalid Credentials Error: {error} Message: {message}")
```

Comment:
be carefull with switching naming here, the fields in the payload are called 'error' and 'error_description', changing it to message can make it super hard to debug if we want to replicate it in curl or something. Try to keep it as close to the original message as possible.

Also, we are now doing error parsing in two different locations, there's the parse_error method that is handling api errors, and then this that handles only the oauth stuff. We are now introducing a third type of error, that is not a pardot error and it makes things a bit complicated.

Reply:
In regards to the message formatting: I'll just print the `data` object in the message. 

Regarding the extra error type: 
- [1] It was actually the fact that we are trying to access `status_code` in `data` object that was blowing up (given that, in case of failure when fetching token, it doesn't have the `status_code` field). Since it blew there, the error was 100% lost in the logs. There is a workaround, I could use `data.get()` and check for existence but I don't think it is worth it due to [2].
```python
def parse_error(response: requests.Response) -> Tuple[str, int]:
    error: str
    code: int
    if response.headers.get("content-type") != "application/json":
        code = response.status_code
        error = "PardotAPIError: " + response.text
    else:
        data: Dict = response.json()
        code = cast(int, data.get("@attributes", {}).get("err_code"))
        error = cast(str, data.get("err"))

    return error, code
``` 
- [2] `PardotException`, the way it is built right now, it expects a response coming from pardot that has the error format that pardot defines in their API. Notice that when a we fail to get an access_token we do not get that kind of error, but a proper http error.

- [3] I'll need to catch the exception up the flow. It is very handy to have a concrete exception that will result in exit(5)